### PR TITLE
refactor(blindspot): replace pass judge with arc length comparison

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Description

- [x] refactor to remove complexity

verify behavior is same as before
note: pass judge insertion can add more complexity because it needs to handle stop line index if pass judge is inserted.
image

```
  /* insert judge point */
  // need to remove const because pass judge idx will be changed by insert stop point
  int pass_judge_idx_ip = std::min(
    static_cast<int>(path_ip.points.size()) - 1, std::max(stop_idx_ip - pass_judge_idx_dist, 0));
  if (has_prior_stopline || stop_idx_ip == pass_judge_idx_ip) {
    *pass_judge_line_idx = *stop_line_idx;
  } else {
    bool is_pass_judge_point_inserted = true;
    //! insertPoint check if there is no duplicated point
    *pass_judge_line_idx =
      insertPoint(pass_judge_idx_ip, path_ip, path, is_pass_judge_point_inserted);
    if (is_pass_judge_point_inserted)
      ++(*stop_line_idx);  // stop index is incremented by judge line insertion
  }
```
comparing to arclength pass judge method so change not to insert pass judge point.
https://user-images.githubusercontent.com/65527974/186405257-b8c0c3f2-a2b7-4232-b537-82c1b2e7a63c.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp [hunk 1]
```diff
@@ -110,21 +109,32 @@ bool BlindSpotModule::modifyPathVelocity(
...
   const size_t closest_idx = closest_idx_opt.get();
 
+  /* set judge line dist */
+  const double current_vel = planner_data_->current_velocity->twist.linear.x;
+  const double max_acc = planner_data_->max_stop_acceleration_threshold;
+  const double delay_response_time = planner_data_->delay_response_time;
+  const double pass_judge_line_dist =
+    planning_utils::calcJudgeLineDistWithAccLimit(current_vel, max_acc, delay_response_time);
+  const auto stop_point_pose = path->points.at(stop_line_idx).point.pose;
+  const auto distnce_until_stop = motion_utils::calcSignedArcLength(
```
Comment (purewater0901) [inline planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp:119]: Spell Error `distance_until_stop`

File: planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp [hunk 2]
```diff
@@ -110,21 +109,32 @@ bool BlindSpotModule::modifyPathVelocity(
...
+    input_path.points, current_pose.pose, stop_point_pose.position);
+  if (distnce_until_stop == boost::none) return true;
+
   /* get debug info */
   const auto stop_line_pose = planning_utils::getAheadPose(
     stop_line_idx, planner_data_->vehicle_info_.max_longitudinal_offset_m, *path);
   debug_data_.virtual_wall_pose = stop_line_pose;
-  debug_data_.stop_point_pose = path->points.at(stop_line_idx).point.pose;
-  debug_data_.judge_point_pose = path->points.at(pass_judge_line_idx).point.pose;
+  const auto stop_pose = path->points.at(stop_line_idx).point.pose;
```
Comment (purewater0901) [inline planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp:127]: Isn't it same as the 'stop_line_pose'

Comment (taikitanaka3) [reply planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp:127]: @purewater0901 
no it's stop pose for ego vehicle

File: planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp (hunk 1 above)
Comment (taikitanaka3) [reply planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp]: I fixed at [621b2c0](https://github.com/autowarefoundation/autoware.universe/pull/1682/commits/621b2c046c1acc93e4995dcd09486a71ef8baa06)

File: planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp (hunk 2 above)
Comment (purewater0901) [reply planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp:127]: okay. got it

**Review by purewater0901 — APPROVED** [review]
LGTM

**Pull request merged**